sdk/trace/time.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"time"
+)
+
+// Clock is responsible for providing time for span's start/end timestamp.
+// By default the standard "time" package will be used. User can replace
+// it with customized clock implementation (e.g. has additional clock
+// synchronization logic) by using the `WithClock` option.
+type Clock interface {
+	Now() time.Time
+}
+
+type clockWrapper struct {
+	NowFunc   func() time.Time
```

Comment:
I don't think NowFunc or SinceFunc need to be exported.

Reply:
Indeed. Code updated. @dashpole